### PR TITLE
Chengyan phase2 update consumables bulk

### PR DESCRIPTION
**Pull request by ChengyanOo**
# Description
This feature is part of HGN app Phase Two, incorporating UI implementations and Redux integrations to support bulk update functionality for consumables.

## Related PRs:
[HGNRest_PR#736](https://github.com/OneCommunityGlobal/HGNRest/pull/736/)

## Main Changes Explained:
- Introduced new components `UpdateConsumablesBulk.css`, `UpdateConsumablesBulk.jsx`, `UpdateConsumablesBulkInputs.jsx`, `UpdateConsumablesBulkTable.jsx`, `UpdateConsumable.css`, and `UpdateConsumable.jsx` to facilitate the bulk update feature for consumables.
- Added an endpoint connection to `/bm/updateConsumableRecordBulk` REST endpoint.
- Enhanced Redux store with actions and reducers to manage the bulk update functionality.
- Corrected typos in the `updateMaterialRecordBulk` reducers.

## How to Test:
To test these changes, follow these steps:
1. Switch to the [Chengyan-Phase2-Update-Consumables-Bulk-Backend](https://github.com/OneCommunityGlobal/HGNRest/tree/Chengyan-Phase2-Update-Consumables-Bulk-Backend) backend branch and start the server.
2. Check out the current branch.
3. Execute `npm install` and `npm run start:local` to run this PR locally.
4. Clear site data/cache.
5. Log in as an admin user.
6. Navigate to the BM Dashboard → Consumables → Update at [http://localhost:3000/bmdashboard/consumables/update](http://localhost:3000/bmdashboard/consumables/update).
7. Modify the record on the form and verify if the processes make sense. You can go to `/bmdashboard/consumables` to verify the consumables data.
8. Review the code for potential code smells.

## Screenshots or Videos of Changes:
Screenshot of the bulk update feature

For a detailed walkthrough, visit:
[https://www.loom.com/share/2576737e00bf4ef38c4ae72c46896d52?sid=11a8baf4-51f0-4f90-a41f-3103fae06d6c](https://www.loom.com/share/2576737e00bf4ef38c4ae72c46896d52?sid=11a8baf4-51f0-4f90-a41f-3103fae06d6c)

### CODE REVIEW

File: src/components/BMDashboard/Consumables/UpdateConsumables/UpdateConsumablesBulk/UpdateConsumablesBulkInputs.jsx
Comment (nnamdixobi) [inline src/components/BMDashboard/Consumables/UpdateConsumables/UpdateConsumablesBulk/UpdateConsumablesBulkInputs.jsx:1]: well named and appropriately organized to facilitate the bulk update feature

File: src/components/BMDashboard/Consumables/UpdateConsumables/UpdateConsumablesBulk/UpdateConsumablesBulk.jsx
Comment (nnamdixobi) [inline src/components/BMDashboard/Consumables/UpdateConsumables/UpdateConsumablesBulk/UpdateConsumablesBulk.jsx:1]: well named and appropriately organized to facilitate the bulk update feature.

**Review by XiaohanMeng — DISMISSED** [review]
Hi, I successfully tested this PR and everything functioned as intended—great job! Here's how I conducted the tests:
* Verified the submit button functionality with all "used" counts at 0.
* Checked the submit button when the "used" count exceeded the available limit.
* Ensured data updates correctly after submission.
Please refer the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/5d999767-2e9f-4bf0-9bee-5396e7e8c450

**Review by brandta-1 — DISMISSED** [review]
# Review
Successfully changed consumable availability. Only point to note is that the New Available for Building 3 Batteries said 29.1537 in the update page, but said 29.1538 in the updated consumables page, but this may just be an insignificant rounding.

# Video

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/116298512/161136e3-f7aa-4432-b51e-3eab89ba797a

**Review by sucheta90 — CHANGES_REQUESTED** [review]
Hello @ChengyanOo I tested this PR and followed the steps mentioned to test this PR, the submit button worked as intended, the calculations in the update table works as intended, however when I tried to submit the, I got a TypeError. Please refer to the 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/8dd28a7e-edd5-4dc5-b5b0-fce1fcbcf244

**Review by Sandhya1236 — CHANGES_REQUESTED** [review]
I attempted to test the new bulk update functionality for consumables as part of HGN app Phase Two. This feature includes new components (UpdateConsumablesBulk.css, UpdateConsumablesBulk.jsx, UpdateConsumablesBulkInputs.jsx, UpdateConsumablesBulkTable.jsx, UpdateConsumable.css, and UpdateConsumable.jsx), an endpoint connection to /bm/updateConsumableRecordBulk, and enhancements to the Redux store. After switching to the Chengyan-Phase2-Update-Consumables-Bulk-Backend branch and starting the server, I checked out the current branch, executed npm install, and ran the application locally. However, I encountered a compilation error due to an unresolved merge conflict in the URL.js file. The conflict markers (<<<<<<< HEAD, =======, and >>>>>>>) indicate incompatible changes that need to be manually resolved. As a result, I could not test the bulk update functionality for consumables successfully.

image

**Pull request closed without merging**